roxie/ccd/ccddali.cpp

```diff
@@ -436,21 +436,38 @@ class CRoxieDaliHelper : implements IRoxieDaliHelper, public CInterface
     IMPLEMENT_IINTERFACE;
     CRoxieDaliHelper() : serverStatus(NULL), connectWatcher(this)
     {
-        userdesc.setown(createUserDescriptor());
-        const char *roxieUser = NULL;
-        const char *roxiePassword = NULL;
-        if (topology)
-        {
-            roxieUser = topology->queryProp("@ldapUser");
-            roxiePassword = topology->queryProp("@ldapPassword");
-        }
-        if (!roxieUser)
-            roxieUser = "roxie";
-        if (!roxiePassword)
-            roxiePassword = "";
-        StringBuffer password;
-        decrypt(password, roxiePassword);
-        userdesc->set(roxieUser, password.str());
+        if (oneShotRoxie && topology && topology->hasProp("@workunit"))
+        {
+            connect(ROXIE_DALI_CONNECT_TIMEOUT);
+            if (!connected())
+                throw MakeStringException(ROXIE_DALI_ERROR, "Failed to connect to dali");
+            Owned<IWorkUnitFactory> wuFactory = getWorkUnitFactory();
+            Owned<IWorkUnit> w = wuFactory->updateWorkUnit(topology->queryProp("@workunit"));
+            if (!w)
+                throw MakeStringException(ROXIE_DALI_ERROR, "Failed to locate dll workunit info");
+            w->setAgentSession(myProcessSession());
+            if(topology->getPropBool("@resetWorkflow", false))
+                w->resetWorkflow();
+            userdesc.set(w->queryUserDescriptor());
+        }
```

Comment:
should possibly have w->commit() so that no exceptions can be thrown from the destructor.